src/iam/client/iamclient.cpp

```diff
@@ -0,0 +1,316 @@
+/*
+ * Copyright (C) 2024 Renesas Electronics Corporation.
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "iamclient.hpp"
+#include "grpchelper.hpp"
+#include "iam/log.hpp"
+
+namespace iam {
+
+/***********************************************************************************************************************
+ * Public
+ **********************************************************************************************************************/
+
+aos::Error RemoteIAMClient::Init(const Config& config, aos::iam::certhandler::CertHandlerItf& certHandler,
+    aos::cryptoutils::CertLoaderItf& certLoader, aos::crypto::x509::ProviderItf& cryptoProvider, bool provisioningMode)
+{
+
+    if (provisioningMode) {
+        mCredetials = grpc::InsecureChannelCredentials();
+    } else {
+        aos::iam::certhandler::CertInfo certInfo;
+
+        auto err = certHandler.GetCertificate(aos::String(config.mCertStorage.c_str()), {}, {}, certInfo);
+        if (!err.IsNone()) {
+            LOG_ERR() << "Get certificates failed, error = " << err.Message();
+
+            return AOS_ERROR_WRAP(aos::ErrorEnum::eInvalidArgument);
+        }
+
+        mCredetials = GetTlsChannelCredentials(certInfo, certLoader, cryptoProvider);
+    }
+
+    for (const auto& iamCfg : config.mRemoteIAMs) {
+        mRemoteIMs[iamCfg.mNodeID.c_str()] = {iamCfg, nullptr};
+    }
+
+    if (mRemoteIMs.size() > cMaxNodes) {
+        mRemoteIMs.clear();
+
+        return AOS_ERROR_WRAP(aos::ErrorEnum::eNoMemory);
+    }
+
+    return aos::ErrorEnum::eNone;
+}
+
+aos::Array<aos::StaticString<aos::cNodeIDLen>> RemoteIAMClient::GetRemoteNodes()
+{
+    aos::StaticArray<aos::StaticString<aos::cNodeIDLen>, cMaxNodes> result;
+
+    std::lock_guard lock(mMutex);
+
+    for (const auto& pair : mRemoteIMs) {
+        result.PushBack(pair.first.c_str());
+    }
+
+    return result;
+}
+
+aos::Error RemoteIAMClient::GetCertTypes(
+    const aos::String& nodeID, aos::Array<aos::StaticString<aos::iam::certhandler::cCertTypeLen>>& certTypes)
+{
+    auto stub = CreateIAMProvisioningServiceStub(nodeID);
+    if (!stub) {
+        return AOS_ERROR_WRAP(aos::ErrorEnum::eFailed);
+    }
+
+    iamanager::v4::GetCertTypesRequest request;
+    request.set_node_id(nodeID.CStr());
+
+    iamanager::v4::CertTypes response;
+
+    grpc::ClientContext ctx;
+    SetClientContext(ctx, nodeID);
+
+    if (const auto status = stub->GetCertTypes(&ctx, request, &response); !status.ok()) {
+        LOG_DBG() << "GetCertTypes failed. error_message = " << status.error_message().c_str();
```

Comment:
Do not use function names or other internals name in log :`Getting cert types failed: error = ...`